packages/hebao_v1/contracts/modules/security/GuardianUtils.sol

```diff
@@ -53,66 +48,43 @@ library GuardianUtils
             if (signers[i] == walletOwner) {
                 walletOwnerSigned = true;
             } else {
-                require(securityStore.isGuardian(wallet, signers[i]), "SIGNER_NOT_GUARDIAN");
-                signingGuardians[numGuardians++] = securityStore.getGuardian(wallet, signers[i]);
+                require(isWalletGuardian(allGuardians, signers[i]), "SIGNER_NOT_GUARDIAN");
             }
         }
 
+        if (walletOwnerSigned) {
+            totalNumVotes += 1;
```

Comment:
I think we need to do this instead for `totalNumVotes`: 
```
if (requirement != SigRequirement.OwnerNotAllowed) {
    totalNumVotes += 1;
}
```

Reply:
I think owner should not included in totalNumVotes if it not in the signers. In this way we need less guardian signers in some case.

Reply:
Certainly not a big difference, only a difference when `OwnerAllowed` is used. With the current code, for example if a wallet only has a single guardian and `OwnerAllowed` is enabled, only that singly guardian can sign (and the wallet owner can also sign, but then that single guardian also still needs to sign because then suddenly 2 signatures are required so no reason to do that). Maybe a bit unexpected.

Reply:
If a wallet only has a single guardian, then that guardian can do whatever he want to the wallet, so nothing unexpected. :)

Reply:
Oh sure, but similar cases with a different number of guardians.

I guess what I find unexpected is that there's no fixed number of signatures required and the signature of the wallet owner is somehow worth less than a signature of a guardian. I'm unsure if that makes sense or not.

On the other hand, we don't really use `OwnerAllowed` any more I think so it doesn't really matter now. :)

Reply:
`OwnerAllowed` is not really used but I feel like have it for future usage :)

Reply:
@Brechtpd I have improved the majority calculation code based on the tables in https://github.com/Loopring/protocols/issues/1924, please take another look at this PR.